std/utf.d

```diff
@@ -2729,7 +2729,8 @@ private P toUTFzImpl(P, S)(S str) @safe pure
     {
         typeof(*P.init)[] retval = ['\0'];
 
-        return retval.ptr;
+        auto trustedPtr() @trusted { return retval.ptr; }
+        return trustedPtr();
```

Comment:
Perhaps it would be good practice to rename the enclosing function `trustedToUTFzImpl` because we're escaping `retval.ptr`. This makes it clear that the function's interface is not safe, even though it is not marked @system.